### ISSUE TITLE
null-ls is unreliable - solution proposal

### ISSUE THREAD

**Issue by Zeioth** [Zeioth/garbage-day.nvim#2]
We are currently storing the stopped buffers this way

```lua
for _, client in pairs(vim.lsp.get_active_clients()) do
  for buf, _ in pairs(client.attached_buffers) do
    stopped_lsp_clients =  ----
```

Which it works. But I've noticed in order for null-ls to start reliably, the current language LSP server must be running first. This makes the order in which we store the stopped clients important. So I propose the next solution:

```lua
for _, buf in ipairs(vim.api.nvim_list_bufs()) do
  for _, client in ipairs(vim.lsp.get_clients()) do
    stopped_lsp_clients =  ----
```

Even though this code is slightly less intuitive, it should ensure the buffers are stored in the order they were originally started. I'll be testing this solution tomorrow.

**Comment by Zeioth** [comment]
Nah, nothing we can do. This is just null-ls not attaching stuff, or having timeout. All other lsp clients work as expected.

All we can do is adding a option to ignore certain lsp client (which is what null-ls is).

**Comment by Zeioth** [comment]
Fixes c1eb3125e1ed1eed31be3e394a52fe5d652bd63e
Closing.
